src/elements/image/ImageElementForm.tsx

```diff
@@ -53,34 +55,64 @@ const htmlLength = (text: string) => {
   return el.innerText.length;
 };
 
+export const thumbnailOption = {
+  text: "thumbnail",
+  value: "thumbnail",
+};
+
 export const ImageElementForm: React.FunctionComponent<Props> = ({
   errors,
   fields,
   fieldValues,
+  roleOptionsStore: RoleOptionsStore,
 }) => {
   return (
     <div data-cy={ImageElementTestId}>
       <Columns>
         <Column width={2 / 5}>
           <FieldLayoutVertical>
-            <CustomDropdownView
-              field={fields.role}
-              label="Weighting"
-              errors={errors.role}
-              options={
-                minAssetValidation(fieldValues.mainImage, "").length
+            <RoleOptionsStore>
+              {(additionalRoleOptions) => {
+                const roleOptions = minAssetValidation(
+                  fieldValues.mainImage,
+                  ""
+                ).length
                   ? [thumbnailOption]
-                  : undefined
-              }
-            />
+                  : [...additionalRoleOptions, thumbnailOption];
+
+                /**
+                 * We must check our role when our role options change, to
+                 * ensure that the value we've selected exists within our list
+                 * of options. If it doesn't, we update our field value to the
+                 * first available option.
+                 */
+                useEffect(() => {
+                  const roleHasBeenRemoved = !roleOptions.some(
+                    ({ value }) => value === fieldValues.role
+                  );
+
+                  if (roleHasBeenRemoved) {
+                    fields.role.update(roleOptions[0].value);
+                  }
+                }, [roleOptions]);
```

Comment:
Not an ideal solution, but should do the job!

Reply:
Haha, yeah 😅